File: tests/CoreTests/Properties_/CustomPropertyName.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SharpKit.JavaScript;

namespace CoreTests.Properties_
{
    [JsType(JsMode.Prototype)]
    class CustomPropertyName
    {
        public CustomPropertyName()
        {
            this.CSharpName = "Bar";
        }
        [JsProperty(Name = "jsName")]
        public JsString CSharpName { get; private set; }
    }
}

File: trunk/tests/CoreTests/Properties_/CustomPropertyName.js
if (typeof(CoreTests) == "undefined")
    var CoreTests = {};
if (typeof(CoreTests.Properties_) == "undefined")
    CoreTests.Properties_ = {};
CoreTests.Properties_.CustomPropertyName = function ()
{
    this._jsName = null;
    this.set_jsName("Bar");
};
CoreTests.Properties_.CustomPropertyName.prototype.get_jsName = function ()
{
    return this._jsName;
};
CoreTests.Properties_.CustomPropertyName.prototype.set_jsName = function (value)
{
    this._jsName = value;
};
